packages/app/obojobo-express/models/edit_lock.js

```diff
@@ -0,0 +1,87 @@
+const db = require('../db')
+const editLockExpireMinutes = oboRequire('config').general.editLockExpireMinutes
+
+class EditLock {
+	constructor(props) {
+		// expand all the props onto this object
+		for (const prop in props) {
+			this[prop] = props[prop]
+		}
+	}
+
+	// returns count of all rows deleted
+	static deleteExpiredLocks() {
+		return db.result(
+			`
+			DELETE
+			FROM edit_locks
+			WHERE created_at < now() - interval '${editLockExpireMinutes} minutes'
+				`,
+			null,
+			r => r.rowCount // extract rowCount from query data
+		)
+	}
+
+	// clear a user's locks on a draft
+	// returns true if deleted, false if nothing to delete
+	static deleteByDraftIdandUser(userId, draftId) {
```

Comment:
It seems that this isn't being called, however I think we should as a `onbeforeunload` in the editor. We can use `navigator.sendBeacon` to post to an API end point to explicitly delete a lock when we know the user is stopping editing. We could potentially also have a `deleteByUser` which could be called on logout?

Reply:
Do we also need to call this one someone loses permissions to an object? If you start editing a module, then get your access to that module removed the lock remains and the original owner is stuck waiting for the lock to expire.

Reply:
All good points, but I think having minimal impact right now, perhaps v2 of locking?